components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/util/Utils.java

```diff
@@ -361,4 +363,40 @@ public static boolean isOrganizationQualifiedURLsSupported(String organizationId
         }
         return true;
     }
+
+    public static int getTenantId(String tenantDomain) throws RuntimeException {
+
+        int tenantId = MultitenantConstants.INVALID_TENANT_ID;
+        try {
+            if (OrganizationManagementDataHolder.getInstance().getRealmService() != null) {
+                tenantId = OrganizationManagementDataHolder.getInstance().getRealmService().getTenantManager()
+                        .getTenantId(tenantDomain);
+            }
+        } catch (UserStoreException e) {
+            throw new RuntimeException("Error occurred while retrieving tenantId for tenantDomain: " + tenantDomain +
+                    e.getMessage(), e);
+        }
+        if (tenantId == MultitenantConstants.INVALID_TENANT_ID) {
+            throw new RuntimeException("Invalid tenant domain " + tenantDomain);
+        } else {
+            return tenantId;
+        }
+    }
+
+    public static String getTenantDomain(int tenantId) throws RuntimeException {
+
+        String tenantDomain = null;
```

Comment:
don't we need to handle tenantId = -1234 case?

Reply:
This is also handled by the TenantManager which is invoked inside this method.